app/controllers/invitation_controller.rb

```diff
@@ -74,47 +82,53 @@ def accept
     inviter_user_id = @inv.from_id
     inviter_participant = AssignmentParticipant.find_by_user_id_and_assignment_id(inviter_user_id, assignment_id)
 
-    ready_to_join = false
-    # check if the inviter's team is still existing, and have available slot to add the invitee
-    inviter_assignment_team = AssignmentTeam.team(inviter_participant)
-    if inviter_assignment_team.nil?
-      flash[:error] = "The team that invited you does not exist anymore."
-    else
-      if inviter_assignment_team.full?
-        flash[:error] = "The team that invited you is full now."
-      else
-        ready_to_join = true
-      end
-    end
-
-    if ready_to_join
-      @inv.reply_status = 'A'
-      @inv.save
+    
+    if ready_to_join?(inviter_participant)
+	    @inv.reply_status = 'A'
+	    @inv.save
 
-      # Remove the users previous team since they are accepting an invite for possibly a new team.
-      TeamsUser.remove_team(student.user_id, params[:team_id])
+	    # Remove the users previous team since they are accepting an invite for possibly a new team.
```

Comment:
I had the wrong set up for indentation in vim and it didn't follow ruby custom...